internal/olm/manager.go

```diff
@@ -96,16 +93,19 @@ func (m *Manager) Uninstall() error {
 	ctx, cancel := context.WithTimeout(context.Background(), m.Timeout)
 	defer cancel()
 
-	version, err := m.Client.GetInstalledVersion(ctx, m.OLMNamespace)
-	if err != nil {
-		return err
+	if m.Version == "" {
+		version, err := m.Client.GetInstalledVersion(ctx, m.OLMNamespace)
```

Comment:
I can set `--version=0.14.0` when the running OLM version is `0.15.1` and it works without reporting an error. Can you rewrite this like:
```go
if version, err := m.Client.GetInstalledVersion(ctx, m.OLMNamespace); err != nil {
	// Return an error if the version override is not set.
	if m.Version == "" {
		return fmt.Errorf("error getting installed OLM version (set --version to override this error): %v", err)
	}
} else if m.Version != "" {
	// Uninstalling from a different version's manifests can cause errors.
	// Return an error in this scenario.
	if version != m.Version {
		return fmt.Errorf("mismatched installed version %q vs. supplied version %q", version, m.Version)
	}
} else {
	m.Version = version
}
```
in both `uninstall` and `status`.

Reply:
Makes sense. Updated.